includes/internal/emails/class-email-customization.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * File containing the Email_Customization class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Internal\Emails;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Email_Customization
+ *
+ * @internal
+ *
+ * @since $$next-version$$
+ */
+class Email_Customization {
+
+	/**
+	 * Class instance.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Email_Customization constructor.
+	 *
+	 * Prevents other instances from being created outside of `self::instance()`.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Get the singleton instance.
+	 *
+	 * @internal
+	 *
+	 * @return self
+	 */
+	public static function instance(): self {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initialize the class and add hooks.
+	 *
+	 * @internal
+	 */
+	public function init(): void {
+		( new Email_Post_Type() )->init();
+		( new Settings_Menu() )->init();
+		( new Email_Settings_Tab() )->init();
```

Comment:
I believe it's a good practice to preserve the instances of the sub-objects if they use hooks. For example, having access to the settings menu instance like this: `Email_Customization::instance()->settings_menu`. The benefit of this architecture is that we (or third parties) are able to modify the hooks attached to those instances (using `remove_action` for example).

What do you think of trying to follow this convention?

Reply:
I think you're totally right about the need to preserve instances.
The only thing I can't agree is about using Singleton for that purpose: it puts those objects in global state and it makes testing harder. 
What do you think if I keep them as instance properties?

Reply:
I'm going to merge the PR and address this issue in a separate PR to unblock other cards, but feel free to answer here :)

Reply:
> What do you think if I keep them as instance properties?

Yep, that would be perfect. We can keep only the `Email_Customization` as a singleton and everything else could be kept as instance properties. Maybe something similar to the [glossary](https://github.com/Automattic/sensei-pro/blob/3e9cf6b3ca67020c9a11b7312420a54174804171/modules/glossary/includes/class-glossary.php#L91-L92)?

Reply:
[Yep, did it similar way :) ](https://github.com/Automattic/sensei/pull/6501)